src/Lykke.Job.CandlesProducer.Services/Trades/Spot/SpotTradesSubscriber.cs

```diff
@@ -35,123 +34,88 @@ public class SpotTradesSubscriber : ITradesSubscriber
 
         public void Start()
         {
-            _marketTradesSubscriber = _subscribersFactory.Create<MarketTradesMessage>(_tradesSubscriptionSettings.ConnectionString, "lykke", _tradesSubscriptionSettings.EndpointName, ProcessMarketTradesAsync);
-            _limitTradesSubscriber = _subscribersFactory.Create<LimitTradesMessage>(_tradesSubscriptionSettings.ConnectionString, "lykke", "limitorders.clients", ProcessLimitTradesAsync);
+            _limitTradesSubscriber = _subscribersFactory.Create<LimitOrdersMessage>(_tradesSubscriptionSettings.ConnectionString, "lykke", "limitorders.clients", ProcessLimitTradesAsync);
         }
 
-        private async Task ProcessMarketTradesAsync(MarketTradesMessage message)
+        private async Task ProcessLimitTradesAsync(LimitOrdersMessage message)
         {
-            if (message.Trades == null || !message.Trades.Any())
+            if (message.Orders == null || !message.Orders.Any())
             {
                 return;
             }
 
-            var trades = message
-                .Trades
-                .Select(t =>
+            var limitOrderIds = message.Orders
+                .Select(o => o.Order.Id)
+                .ToHashSet();
+
+            foreach (var orderMessage in message.Orders)
+            {
```

Comment:
limitOrderIds can be added here one by one. Minus one loop over Orders

Reply:
No, they can't. If we do so, then durin processing trades of not last order, limitOrderIds.Contains(tradeMessage.OppositeOrderId) will return false even if given order ID is actually exists in the message.

Reply:
Agree